lib/smart_answer/money.rb

```diff
@@ -9,14 +9,11 @@ class Money
     attr_reader :value
 
     def initialize(raw_input)
-      if raw_input.is_a?(Numeric)
-        @value = BigDecimal.new(raw_input.to_s)
+      number, error_message = self.class.raw_input_to_number(raw_input)
```

Comment:
I'm not sure I like this pattern of returning a tuple with an error message. Why not just `raise` directly from `raw_input_to_number`? It should hopefully tidy up your control logic a lot more.